cppapi/client/zmqeventconsumer.cpp

```diff
@@ -1764,8 +1764,8 @@ void ZmqEventConsumer::connect_event_system(string &device_name,string &obj_name
         ::strcpy(&(buffer[length]),endpoint.c_str());
         length = length + endpoint.size() + 1;
 
-        ::strcpy(&(buffer[length]), received_from_admin.event_name.c_str());
-        length = length + received_from_admin.event_name.size() + 1;
+        ::strcpy(&(buffer[length]), full_event_name.c_str());
```

Comment:
Well, I think this will brake what is done in #423. I have created a new PR (#468), because I can not modify your repo.  This one will be closed, so please continue your work in that PR.